RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc

```diff
@@ -39,6 +48,11 @@ FTLRecHit MTDRecHitAlgo::makeRecHit(const FTLUncalibratedRecHit& uRecHit, uint32
   float energy = 0.;
   float time = 0.;
 
+  // MTD topology
+  unsigned int index_topology = 0;  //1Disks geometry
+  if (topology_->getMTDTopologyMode() > static_cast<int>(MTDTopologyMode::Mode::barphiflat))
```

Comment:
why not `== btlv1etlv4` (with whatever appropriate casts). I do not see any design constraints that would prevent defining another value in `MTDTopologyMode::Mode`, which would correspond again to index 0.

Please remind me how do we get `barphiflat` vs `btlv1etlv4` vs some other value of `getMTDTopologyMode`?
It looks like a property of the full detector (mixing of different values is not possible).
Ideally, then the threshold should come from ES in a coherent way.
Having both values in the same configuration is a bit confusing, because only one can possibly be applicable.
The alternative to using ES is an appropriate Era modifier.
Only if the options above are rejected, I'd consider injecting multiple incompatible caibration/configuration parameters into the same module.

Reply:
@slava77 up to barphiflat (i.e. 4) this enum was used to distinguish various BTL scenarios, now all gone but the last one. I use the same enum now to distinguish ETL scenarios, btlv2etlv4 is the first implementation in D53, but another one is about to be stored as well. This value comes with the MTD topology object, and should be considered as a temporary tool to identify within the code which geometry is being processed, as soon as the choice will be done this will not be needed any more.

In the specific case of this parameter, there is a change in thickness of the silicon detector in the change of scenarios that justify this move, and this change will not be a function of the new scenarios (i.e. LGAD thickness should be fixed now regardless of the geometrical arrangement of modules on the discs' faces). So for what we know now, this selection is ok. Your suggestion, if I understand correctly, would be to create an EStype object that provides these parameters as a function of the topology mode, do I understand it correctly? @casarsa @gsorrentino18 do you see other parameters that need to be tuned?

Reply:
> up to barphiflat (i.e. 4) this enum was used to distinguish various BTL scenarios, now all gone but the last one. I use the same enum now to distinguish ETL scenarios, btlv2etlv4 is the first implementation in D53, but another one is about to be stored as well.

I'm not sure if this historical precedent can suggest that in the future any new enum will also be about ETL 2-disk cases; that's why I think that selection of values explicitly is more appropriate than a range comparison `>`.

> Your suggestion, if I understand correctly, would be to create an EStype object that provides these parameters as a function of the topology mode, do I understand it correctly?

That would be an ideal case; it's probably not much complicated because at least in this case the related data is already handled via ES.
An era modifier is another option.

Reply:
> @casarsa @gsorrentino18 do you see other parameters that need to be tuned?

@fabiocos In principle, yes, the calibration constant (the ETL energy is converted into MIP units in the simulation and the MIP peak value depends on the thickness of the sensor), but we are planning on tackling that in another dedicated PR that modifies also the simulation code.
As for this PR, the calibration constant is an internal parameter (and currently kind of arbitrary) and as long as its value is consistent between simulation and reconstruction the result is independent of its actual value.

Reply:
> That would be an ideal case; it's probably not much complicated because at least in this case the related data is already handled via ES.
> An era modifier is another option.

Following your suggestions, we are indeed thinking about implementing an Era Modifier to provide the parameters as a function of the geometry.

Reply:
@slava77 Before modifying the PR, if I understand well, do you want me to implement a modifier to the Phase2 Era similar to those listed here? https://cmssdt.cern.ch/lxr/source/Configuration/Eras/python/Era_Phase2_cff.py?v=CMSSW_11_2_X
Or were you suggesting something different?

Reply:
> @slava77 Before modifying the PR, if I understand well, do you want me to implement a modifier to the Phase2 Era similar to those listed here? https://cmssdt.cern.ch/lxr/source/Configuration/Eras/python/Era_Phase2_cff.py?v=CMSSW_11_2_X
> Or were you suggesting something different?

yes, that's what I was looking for; I guess you pick the MTD-related name for the modifier and then we see (jointly if it makes sense)
It will be in Era_Phase2CXX_cff file @kpedro88 please suggest a file for this

Reply:
@gsorrentino18 @slava77 it depends on which geometry(s) it covers.

Right now we mainly have top-level Phase2 Eras depending on the calorimeter version (`Phase2C9`, `Phase2C10`, `Phase2C11`, `Phase2C12`). Based on @fabiocos' comments below, the new top-level Era should probably be `Phase2C11I13` (since I13 is the new MTD geometry in #31710). To be concrete, it should start from `Phase2C11` and then add a new modifier like `phase2_etl_2disc`. But MTD experts should correct me if I'm wrong about the context.

Reply:
@kpedro88 @slava77 I am about to submit a PR with the last piece needed to run a whole production workflow with the  new ETL (2 discs), which needs the updates. in this PR (as the thickness of the silicon is now realistically the 50 microns of the LGAD). When this enters in production, and geometry for HGCal provided by @bsunanda should go with it, so this modifier should become in my view an integral part of the baseline Era. I see the special version as a short term interim solution before moving to that new regime, as I expect that the review of the new code will take a bit of time

Reply:
@fabiocos @gsorrentino18 is the suggestion to go ahead with this PR without introducing a new modifier?
if so, there are still a few other unanswered comments; please address these

Reply:
@kpedro88 My intention is to implement an Era Modifier, but since this PR is related to #31765 and #31769, which introduce bigger modifications in the code, I though it would be better to hold this PR until the review of the other two is completed.

Reply:
@kpedro88 @slava77 @bsunanda so what is the preferred strategy? Go for a short period with an intermediate Era and then move the ETL definition into a new Phase2CXX default? Or we may keep this PR on hold and use it for tests of #31765 with #31769, add an Era there and then merge everything in one shot. @silviodonato @qliphy please clarify the preferred strategy

Reply:
@fabiocos I'm ok with having an intermediate Era